mongo_printer.py

```diff
@@ -85,7 +93,18 @@ def to_string(self):
         if size == 5:
             return "%s empty BSONObj @ %s"%(ownership, ptr)
         else:
-            return "%s BSONObj %s bytes @ %s"%(ownership, size, ptr)
+            if bson is not None:
+                inferior = gdb.selected_inferior()
+                buf = bytes(inferior.read_memory(ptr, size))
+                try:
+                    options = CodecOptions(document_class=collections.OrderedDict)
+                    bsondoc = bson.BSON.decode(buf, codec_options=options)
+                    return "%s BSONObj %s bytes @ %s: %s" % (
+                        ownership, size, ptr, pprint.pformat(bsondoc))
```

Comment:
Can you use the display_hint() == 'map' and children() style like StringMap does?